easybuild/tools/docs.py

```diff
@@ -556,6 +556,49 @@ def list_toolchains_txt(tcs):
     return '\n'.join(doc)
 
 
+def avail_toolchain_opts(name, output_format=FORMAT_TXT):
+    """Show list of known options for given toolchain."""
+    tc_class, _ = search_toolchain(name)
+    if not tc_class:
+        return "Couldn't find toolchain: '%s'. To see available toolchains, use --list-toolchains" % name
+    tc = tc_class(version='1.0') # version doesn't matter here, but needs to be defined
+
+    options = [tc.COMPILER_SHARED_OPTS, tc.COMPILER_UNIQUE_OPTS, tc.MPI_SHARED_OPTS, tc.MPI_UNIQUE_OPTS]
+
+    tc_dict = {}
+    for opt in options:
+        if opt is not None:
+            tc_dict.update(opt)
+
+    return generate_doc('avail_toolchain_opts_%s' % output_format, [name, tc_dict])
+
+
+def avail_toolchain_opts_rst(name, tc_dict):
+    """ Returns overview of toolchain options in rst format """
+    title = "Available options for %s toolchain:" % name
+
+    table_titles = ['option', 'description', 'default']
+
+    table_values = [
+        ['``%s``' % opt_name for opt_name in tc_dict.keys()],
+        ['%s' % val[1] for val in tc_dict.values()],
+        ['``%s``' % val[0] for val in tc_dict.values()],
```

Comment:
two more problems here: sorting, and you have no strict guarantee that `.values()` will return things in the same key-order when being called twice

so, collect `.items()` before, and sort by key, then build up `table_values`